### PR TITLE
Add schema for using as a base in LSP plugins

### PR DESCRIPTION
**Pull request by rchl**

### CODE REVIEW

File: sublime-package.json [hunk 1]
```diff
@@ -414,7 +428,29 @@
...
+          "$id": "sublime://settings/LSP-plugin-base",
+          "properties": {
+            "languages": {
+              "$ref": "sublime://settings/LSP#/definitions/ClientLanguages"
+            },
+            "initializationOptions": {
+              "$ref": "sublime://settings/LSP#/definitions/ClientInitializationOptions"
+            },
+            "settings": {
+              "ref": "sublime://settings/LSP#/definitions/ClientSettings"
```
Comment (rwols) [inline sublime-package.json:443]: `"ref"` -> `"$ref"` ?

Comment (rchl) [reply sublime-package.json]: Good catch

**Review by rwols — APPROVED** [review]
I’m worried we’re falling into the fragile base class trap. Consider that the LanguageConfig class may turn into the DocumentSelector class in the future.

**Comment by rchl** [comment]
I don't know, we'd have to figure it out somehow. We can't just change it instantly one day anyway as it will break all plugins. We need to provide some backward compatibility. At least for a certain period of time.

**Pull request merged**